### ISSUE TITLE
Warnings about "bind: Key with name X does not have any mapping" with SSH

### ISSUE THREAD

**Issue by mojoTX** [fish-shell/fish-shell#5372]
I first noticed this issue when I was using `rsync` to copy files from one place to another. I have fish shell set as my default shell on both sides of the connection. 

For instance:
```
[I] mojo@MacBook-Pro ~> rsync -avP mojo@localhost:/tmp/data.raw /tmp/data-copy.raw
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'dc' does not have any mapping
bind: Key with name 'dc' does not have any mapping
receiving incremental file list
data.raw
    268,435,456 100%   87.49MB/s    0:00:02 (xfr#1, to-chk=0/1)

sent 43 bytes  received 268,501,088 bytes  76,714,608.86 bytes/sec
total size is 268,435,456  speedup is 1.00
```

Based on some Google-Fu, this appears to be similar to https://github.com/fish-shell/fish-shell/issues/4431 and https://github.com/fish-shell/fish-shell/issues/4188. 

Normally I use the "oh-my-fish" framework, but I confirmed that that's not the issue, by setting HOME to a random directory. I get this error with plain vanilla fish, with no customization at all. 

I did some more testing, and it appears that OpenSSH doesn't set the TERM environment variable unless there's a pseudo-terminal allocation, and apparently a pseudo-terminal is not allocated for `rsync`. 

With no psuedo-terminal, `TERM` is not set:
```
mojo@MacBook-Pro ~> ssh -T -o VisualHostKey=no mojo@localhost 'env | fgrep TERM '
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'ppage' does not have any mapping
bind: Key with name 'npage' does not have any mapping
bind: Key with name 'f1' does not have any mapping
bind: Key with name 'dc' does not have any mapping
bind: Key with name 'dc' does not have any mapping
```

When I force a pseudo-terminal allocation, I see the TERM environment variable, and there is no warning:
```
mojo@MacBook-Pro ~> ssh -t -o VisualHostKey=no mojo@localhost 'env | fgrep TERM '
TERM=xterm-256color
Connection to localhost closed.
```

This is with fish 2.7.1, on a Mac running macOS Mojave. I'm using iTerm2 v3.2.6 as my terminal program, and I normally use `xterm-256color` as my terminal type. I tried changing my font to the standard Apple "Menlo" font, and I also tried using the standard macOS Terminal app, and neither change made any difference. 

Any suggestions? I'm not 100% convinced that this isn't something weird with my configuration, so I am open to suggestions.

**Comment by faho** [comment]
>Based on some Google-Fu, this appears to be similar to #4431 and #4188.

It is the same thing. Those have been mis-tagged. They haven't been fixed in 2.7.X, only master.

So:

Duplicate of #4188

**Comment by mojoTX** [comment]
FYI, this was completely fixed by wrapping my `.config/fish/config.fish` customizations with:

Connecting with rsync is non-interactive, and some of my customizations were not intended for non-interactive cases. 

```fish
if status is-interactive
    # blah blah blah
end
```
